src/pages/index.js

```diff
@@ -22,96 +34,233 @@ const profileDescriptionInput = profileEditModal.querySelector(
 /* -------------------------------------------------------------------------- */
-function handleImageClick(name, link) {
-  previewImagePopup.open(name, link);
+
+// Render card function
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDelete,
+    handleLikeCard
+  );
+  const cardElement = card.getView();
+  section.addItem(cardElement);
 }
 
-function handleProfileEditSubmit(profileData) {
-  const name = profileData.title;
-  const description = profileData.subheader;
-  user.setUserInfo(name, description);
-  editProfilePopup.close();
+// Handle image click
+function handleImageClick(card) {
+  previewImagePopup.open({ name: card.name, link: card.link });
 }
 
-function handleAddCardSubmit(newCardData, cardListElement) {
-  const name = newCardData.title;
-  const alt = newCardData.title;
-  const link = newCardData.url;
-  renderCard({ name, alt, link });
-  newCardPopup.close();
+// Handle card delete
+function handleDelete(card) {
+  confirmation.open();
+  confirmation.setConfirmSubmit(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        confirmation.close();
+        card._handleDeleteCard();
+      })
+      .catch((err) => console.log(err));
+  });
 }
 
-function createCard(cardData) {
-  return new Card(cardData, "#card-template", handleImageClick).getView();
+// Handle Like click
+function handleLikeCard(card) {
+  if (card._isLiked) {
+    api
+      .removeLike(card._id)
+      .then((res) => {
+        card.updateIsLiked(res.isLiked);
+      })
+      .catch((err) => console.log(err));
+  } else {
+    api
+      .addLike(card._id)
+      .then((res) => {
+        card.updateIsLiked(res.isLiked);
+      })
+      .catch((err) => console.log(err));
+  }
 }
+
 /* -------------------------------------------------------------------------- */
 /*                               Event Listeners                              */
 /* -------------------------------------------------------------------------- */
-//Edit Profile Form
+
+// Edit Profile Form
 profileEditBtn.addEventListener("click", () => {
   const userInput = user.getUserInfo();
-  editProfilePopup.setInputValues({
-    title: userInput.name,
-    subheader: userInput.about,
-  });
-  profileEditFormValidator.resetValidation();
+  profileTitleInput.value = userInput.name;
+  profileDescriptionInput.value = userInput.about;
   editProfilePopup.open();
+  profileEditFormValidator.resetValidation();
 });
 
-//New Card Form
+// New Card Form
 addCardBtn.addEventListener("click", () => {
   newCardPopup.open();
   addCardFormValidator.toggleButtonState();
 });
 
+// Delete Card Form
+const confirmation = new PopupDelete({
+  popupSelector: "#delete-card-modal",
+});
+confirmation.setEventListeners();
+
+// Avatar Edit Form
+avatarEditBtn.addEventListener("click", () => {
+  avatarEditPopup.open();
```

Comment:
Please, add `toggleButtonState` here as well to disable the button if the input is empty on opening